tests/io/test_pyarrow_visitor.py

```diff
@@ -209,9 +209,9 @@ def test_pyarrow_variable_binary_to_iceberg() -> None:
 def test_pyarrow_struct_to_iceberg() -> None:
     pyarrow_struct = pa.struct(
         [
-            pa.field("foo", pa.string(), nullable=True, metadata={"field_id": "1", "doc": "foo doc"}),
-            pa.field("bar", pa.int32(), nullable=False, metadata={"field_id": "2"}),
-            pa.field("baz", pa.bool_(), nullable=True, metadata={"field_id": "3"}),
+            pa.field("foo", pa.string(), nullable=True, metadata={b"PARQUET:field_id": "1", "doc": "foo doc"}),
```

Comment:
```suggestion
            pa.field("foo", pa.string(), nullable=True, metadata={"PARQUET:field_id": "1", "doc": "foo doc"}),
```
Shall we make everything in metadata dict regular string (or byte string) for consistency? I think both regular and byte string can result in correct arrow field metadata.

Reply:
Good one, updated! Everything that coerces to bytes can be used.